lib/mongoid/scroll/base64_encoded_cursor.rb

```diff
@@ -0,0 +1,43 @@
+require 'base64'
+require 'json'
+
+module Mongoid
+  module Scroll
+    # Allows to serializer/deserialize the cursor using RFC 4648
+    class Base64EncodedCursor < Cursor
+      class << self
+        def from_cursor(cursor)
```

Comment:
What's a valid use-case for mutating a cursor into a base64 cursor? If there isn't one, I think this shouldn't exist.

Reply:
The one I described in https://github.com/mongoid/mongoid-scroll/pull/32#issuecomment-1450644272.

The `scroll` methods are yielding a `Cursor`, but I need to call `Base64EncodedCursor#to_s` to return it to my API users so that they can forward it in a next request an I can use it to return the next page.

Reply:
Right, I think this is not ideal from the user POV. Can we have `Feed::Item.desc(:position).limit(5).scroll(klass: Base64EncodedCursor)` or something like that instead?

Reply:
It's what I did in 73bb69449f6745b517087ace46e5f8351212ab2c (explanation in https://github.com/mongoid/mongoid-scroll/pull/28#issuecomment-1440807583 3.a.).

We can't have `scroll(klass: Base64EncodedCursor)` because the signature is `scroll(cursor, options)` so it'll have to be in the second parameter (which means that even if `cursor` is `nil`, we'd have to pass it).
It brings a lot of changes and also questions (what do we do if a cursor is specified but is a different class from `klass`?).
The current PR is already getting bigger because of https://github.com/mongoid/mongoid-scroll/pull/32#discussion_r1125675319, so IMO I should just remove `Base64EncodedCursor.from_cursor` for now if you don't like it, focus on the other discussion (not having `Base64EncodedCursor` inherit from `Cursor`), and create another PR later on to tackle the yielded cursors once this one will be done. Good for you?